lib/router/router.js

```diff
@@ -224,6 +224,10 @@ export default class Router extends EventEmitter {
   }
 
   async prefetch (url) {
+    // We don't add support for prefetch in the development mode.
+    // If we do that, our on-demand-entries optimization won't performs better
+    if (process.env.NODE_ENV === 'development') return
```

Comment:
I wonder if we could use the `dev` provided when setting up the app (server side). Not sure if we pass it through to the client side though, will allow for consistent behaviour not based on NODE_ENV.